src/coincident/search/neon_api.py

```diff
@@ -0,0 +1,234 @@
+"""
+Searching for NEON data with spatial and temporal filters is actually quite difficult.
+I couldn't find any STAC catalogs or code to make this easier
+
+https://data.neonscience.org/data-products/DP3.30024.001
+https://www.neonscience.org/resources/learning-hub/tutorials/neon-api-intro-requests-py
+
+The process goes as follows:
+- Use requests to get some basic metadata for all NEON sites that contain discrete return LiDAR point cloud data
+    - Here, we grab the site code, site name, available months, and product URLs
+    - Note that grabbing the specific collection time for each flight requires
+        a lot more key-value pair digging and i do not believe is worth the effort
+    - We grab the 'centroid' (top left of the center-most tile of footprint) here as well
+        this is because there is no easy way to search on an input geometry and we can
+        use the 'centroid' to make that process easier
+- Subset the resulting gdf on the user-input start and end times
+- Subset that gdf by only including sites that have 'centroids' within 1 degree (~111km)
+    we do this because grabbing the actual footprint geometry is time-consuming.
+    the only way to do this dynamically, at least barring the use of
+    more unnecessary and unique llibraries/packages from what i found,
+    is by making a GET request for each site and then reading in the footprint
+    hosted on storage.googleapis.com
+- Grab the footprint geometry for each site
+    These are hosted on storage.googleapis.com in the form of a shapefile
+    that has all of the complex tile footprints as their own row geometries
+"""
+
+from __future__ import annotations
+
+import warnings
+
+import geopandas as gpd
+import pandas as pd
+import pyogrio
+import requests
+from pandas import Timestamp
+from shapely.geometry import Point
+
+warnings.filterwarnings(
+    "ignore", message=".*Geometry is in a geographic CRS.*", category=UserWarning
+)
```

Comment:
I'd suggest leaving this out, but can add it to an example notebook/docs if it's getting cluttered with warnings